### DISCUSSION TITLE
How do I measure per-method gRPC status codes using Istio?

### DISCUSSION THREAD

**Discussion by DerekTBrown** [istio/istio#52968] [General]
I am trying to create Prometheus metrics for my gRPC services. Ideally, I want per-method error rates (w/status codes). However, it doesn't seem like there is a great way to do this:

- As far as I can tell, the Istio-provided gRPC metrics aren't per-method[^1].
- Using the Envoy gRPC stats filter, I can get per-method metrics, but not status codes. It categorizes everything into success or failure. [^2]

Is there an idiomatic way to do this?

[^1]:https://istio.io/latest/docs/reference/config/metrics/#:~:text=gRPC%20Response%20Status%3A
[^2]: https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/grpc_stats_filter

**Comment by salrashid123** [comment]
@DerekTBrown   incase you're still looking, see [this](https://github.com/salrashid123/grpc_stats_envoy_istio?tab=readme-ov-file#enable-metrics) example.     it should still work (i'm currently tyring to see how to adapt the repo to use [Telemetry](https://istio.io/v1.21/docs/tasks/observability/metrics/telemetry-api/) but haven't figured that out
